config/samples/infrastructure_v1beta1_jujucluster.yaml

```diff
@@ -9,9 +9,62 @@ metadata:
     app.kubernetes.io/created-by: cluster-api-provider-juju
   name: jujucluster-sample
 spec:
+  model:
+    name: jujucluster-sample
+    cloudRegion: Boston
+    config: 
+      juju-http-proxy: "http://squid.internal:3128"
+      apt-http-proxy: "http://squid.internal:3128"
+      snap-http-proxy: "http://squid.internal:3128"
+      juju-https-proxy: "http://squid.internal:3128"
+      apt-https-proxy: "http://squid.internal:3128"
+      snap-https-proxy: "http://squid.internal:3128"
+      apt-no-proxy: "localhost,127.0.0.1,ppa.launchpad.net,launchpad.net"
+      juju-no-proxy: "localhost,127.0.0.1,0.0.0.0,ppa.launchpad.net,launchpad.net,10.0.8.0/24,10.246.154.0/24"
+      logging-config: "<root>=DEBUG"
+      datastore: "vsanDatastore"
+      primary-network: "VLAN_2764"
+      force-vm-hardware-version: "17"
+    constraints:
+      arch: amd64
   # using loadbalancer requires metallb on vsphere
   controllerServiceType: loadbalancer
-  cloudEndpoint: 10.246.152.100
+  credential:
+    credentialSecretName: jujucluster-sample-credential-secret
+    credentialSecretNamespace: default
+  cloud:
+    name: jujucluster-sample
+    type: vsphere
+    endpoint: 10.246.152.100
+    regions:
+      - name: Boston
+        endpoint: 10.246.152.100
+    authTypes:
+      - "userpass"
+  additionalApplications:
+    - charmName: vsphere-integrator      
+      applicationName: vsphere-integrator
+      charmChannel: 1.27/edge
+      charmBase: ubuntu@22.04
+      units: 1
+      config:   
+        datastore: vsanDatastore
+        folder: k8s-crew-root 
+      trust: true
+    - charmName: vsphere-cloud-provider      
+      applicationName: vsphere-cloud-provider
+      charmChannel: 1.27/edge
+      charmBase: ubuntu@22.04
+      units: 0
```

Comment:
it's kinda a shame this doesn't match the normal juju bundle schema

Reply:
it easily could (well mostly easily). having a dynamic key like how the application name is in bundles would probably require a custom yaml unmarshaler, but doable.

Reply:
So far I would say that the API design has been mostly guided by implementation details. Mostly because we're still just trying to get this dang stuff to work.

I think it would be good if we can follow up in the v2 epic with a stronger focus on API design driven by use cases. For now, I think this is okay.

Reply:
since it wasnt a huge change, I went ahead and structured the spec to be more bundle like. Its not a 1 to 1 replacement, but its the same format with applications being provided as a map and relations provided alongside it (with some key differences that relations are now called integrations, and base is now used instead of series)